File: GFG_Problems/Reverse-Bits.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
//User function Template for C++

class Solution
{
    public:
    unsigned int reverseBits(unsigned int n)
    {
        int res = 0;
        
        while(n>0)
        {
            res = res<<1;
            res = res + (n%2);
            n = n>>1;
            
        }
        //code here
        
        return res;
        
    }
};

//{ Driver Code Starts.
int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int N;
        cin >> N;
        
        Solution ob;
        cout<< ob.reverseBits(N) <<"\n";
        
    }
    return 0;
}

// } Driver Code Ends

File: GFG_Problems/reverse-bits.py
#User function Template for python3

class Solution:
    def reversedBits(self, x):
        # code here 
        bins = bin(x)
        bins = list(bins)
        bins.remove('b')
        
        bins = bins[::-1]
        st = ''.join(bins) +'0'*(32-len(bins))
        
        return int(st,2)

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        X=int(input())
        
        ob = Solution()
        print(ob.reversedBits(X))
# } Driver Code Ends
